File: submission_002-mastermind-master/mastermind.py
import random


def run_game():
    code = []
    while (len(code)<4):
        not_duplicate = random.randint(1, 8)
        if (not_duplicate not in code):
            code.append(not_duplicate)

    print("4-digit Code has been set. Digits in range 1 to 8. You have 12 turns to break it.")

    number_of_turns = 12
    while number_of_turns > 0:
        user_input = input("Input 4 digit code: ")

        while user_input.isdigit == False or len(user_input) != 4:
            print("Please enter exactly 4 digits.")
            user_input = input("Input 4 digit code: ")
        correct_place = 0
        not_correct_place = 0

        for count in range(0, len(code)):
            if int(user_input[count]) == code[count]:
                correct_place += 1
            elif int(user_input[count]) in code:
                not_correct_place += 1
            
        
        print("Number of correct digits in correct place:     " + str(correct_place))
        print("Number of correct digits not in correct place: " + str(not_correct_place))
        if correct_place == 4:
            print("Congratulations! You are a codebreaker!" )
            print("The code was: " ,end="")
            for i in range(len(code)):
                print(code[i], end="")
            break
        elif number_of_turns > 1:
            number_of_turns -= 1

            print("Turns left: " + str(number_of_turns))
        elif number_of_turns == 1:
            print("The code was: " ,end="")
            for i in range(len(code)):
                print(code[i], end="")
            break


if __name__ == "__main__":
    run_game()

File: submission-102-java-003-mastermind-master/src/main/java/za/co/wethinkcode/mastermind/Mastermind.java
package za.co.wethinkcode.mastermind;

public class Mastermind {
    private final String code;
    private final Player player;

    public Mastermind(CodeGenerator generator, Player player){
        this.code = generator.generateCode();
        this.player = player;
    }
    public Mastermind(){
        this(new CodeGenerator(), new Player());
    }

    public void runGame(){
        //TODO: implement the main run loop logic
        System.out.println("4-digit Code has been set. Digits in range 1 to 8. You have 12 turns to break it.");
        int chances = 12;
        int guesses = 0;

        String generateCode = code;
        while(guesses != 12){
            int correctPlace = 0;
            int incorrectPlace = 0;
            String playGame = player.getGuess();

            for(int i = 0 ; i < 4; i++){
                char x = generateCode.charAt(i);

                if(playGame.charAt(i) == generateCode.charAt(i)){
                    correctPlace++;
                }else if(playGame.contains(String.valueOf(x))){
                    incorrectPlace++;
                }
            }
            if(correctPlace == 4) {
                System.out.println("Number of correct digits in correct place: " + correctPlace);
                System.out.println("Number of correct digits not in correct place: " + incorrectPlace);
                System.out.println("Congratulations! You are a codebreaker!");
                System.out.println("The code was: " + generateCode);
                break;
            }else{
                System.out.println("Number of correct digits in correct place: " + correctPlace);
                System.out.println("Number of correct digits not in correct place: " + incorrectPlace);
            }
            int chancesLeft = (chances - 1) - guesses;
            if (chancesLeft == 0){
                System.out.println("No more turns left.");
                System.out.println("The code was: " + generateCode);
            }else{
                System.out.println("Turns left: " + ((chances - 1) - guesses));
            }
            guesses++;
        }
    }

    public static void main(String[] args){
        Mastermind game = new Mastermind();
        game.runGame();
//        Player input = new Player();
//        input.getGuess();
    }
}
